maint/local_python/binding_c.py

```diff
@@ -606,6 +741,9 @@ def dump_profiling(func):
     G.out.append("#undef %s" % func_name)
     G.out.append("#define %s P%s" % (func_name, func_name))
     G.out.append("")
+
+    dump_profiling_qmpi(func, func['_map_type'])
+
```

Comment:
This doesn't look right. What should be here is the `QMPI_xxx` function, which is the internal function, i.e. hidden from user, right?  But currently this is `PMPI_xxx`, and `MPI_xxx` is never defined (when weak symbols isn't available). I think this is what causing the failure in this test https://github.com/pmodels/mpich/pull/5035#issuecomment-796365442

Reply:
It is probably easier just add an additional `#ifndef MPICH_MPI_FROM_PMPI` - `#endif` around the `QMPI_xxx` definitons.

Reply:
Actually, shouldn't the `MPI_Send` definition just move out of the `#ifndef MPICH_MPI_FROM_PMPI` block? That way it's defined in all cases?

Reply:
In fact, it would seem like we have the same problem with functions like `MPII_Comm_set_attr`, right? That's defined (apparently intentionally?) inside the `#ifndef MPICH_MPI_FROM_PMPI` block.

Reply:
Yes, `MPII_Comm_set_attr` is similar. But  `MPII_Comm_set_attr` is similar as `QMPI_Comm_set_attr`. So `QMPI_xxx` need moved inside `#ifndef MPICH_MPI_FROM_PMPI` (just as `MPII_Comm_set_attr` does). You are, instead, moving `MPI_xxx` inside the `#ifndef` and leaving the `QMPI_xxx` outside, kinda opposite to what should be done.

Reply:
I'm confused because the QMPI system doesn't rely on the weak symbols tricks that PMPI relies on. the only instance where it should matter is if you have both a QMPI and PMPI tool registered at the same time. I'll see if I can test these cases out, particularly if I can move the MPI function outside of the code block.

My confusion is that if you move the `QMPI` function into the code block, it won't be defined if you do have weak symbols.